### ISSUE TITLE
Does not automatically change between light and dark mode anymore.

### ISSUE THREAD

**Issue by RayDe97** [AutoDarkMode/Windows-Auto-Night-Mode#249]
Not sure if its a bug in the latest Win10 update, but Auto Dark Mode stopped working in the background. When launching the main program, it kicks in and switches themes. Reinstalled multiple time, different install locations, checked task scheduler entries, looks normal. Went through troubleshooting page on GitHub, no dice. Can provide task scheduler log if needed.

Thanks for this program btw, rely on it everyday!

**Comment by Armin2208** [comment]
Hi! Sorry for the late answer and sorry for the outage.

Please provide me your task scheduler log, if possible. Thanks.

Try this version which doesn't use Windows Task Scheduler with new Engine:
[https://www.mediafire.com/file/9xu8wfb3zu2h8v5/AutoDarkModeX_Setup_20212307_1741.exe/file](https://www.mediafire.com/file/9xu8wfb3zu2h8v5/AutoDarkModeX_Setup_20212307_1741.exe/file)

**Comment by RayDe97** [comment]
Unfortunately this version did not work either with specific time. I will try sunset mode, will have to wait till tonight to see. Is there an email address I can send the log to? Or do you want it just here?

**Comment by Armin2208** [comment]
logs just here

**Comment by RayDe97** [comment]
[export.zip](https://github.com/Armin2208/Windows-Auto-Night-Mode/files/7078041/export.zip)

**Comment by Armin2208** [comment]
nice!

**Comment by RayDe97** [comment]
Thanks for your help. Looks like the new version is coming up soon. Looking forward to it, great app! Filling a need it Windows 👍

**Comment by Spiritreader** [comment]
Version X is now out. Please install, try again and let us know if your issue is fixed. If it is, please close the issue.
https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/releases/tag/10.0.0.26

Thank you!

**Comment by RayDe97** [comment]
Everything works without a hitch. Nice!

**Comment by RayDe97** [comment]
![image](https://user-images.githubusercontent.com/77296248/136480108-6cabacf7-3044-4a54-a9ee-36bf4f7a614a.png)
Unfortunately it failed tonight to switch to dark mode at the designated time

**Comment by Spiritreader** [comment]
Please attach the service.log

**Comment by Spiritreader** [comment]
@RayDe97 please switch to the latest beta build (go to settings, select update channel beta and check for updates).
I've added a missing error handler for the GPU module.

Please verify that you are on the right build (at least Version 10.0.1.23 must be indicated on the about page)

After that, please complete the following steps:
- manually set your dark theme time 3 ahead of your current time and let it run to see if it switches now.
For example, if it's currently 10pm, set the time to switch to dark mode to 10:03pm, wait until 10:05pm
- check if the theme switched, and in all cases, attach the service.log again

**Comment by Spiritreader** [comment]
no no, no need to wait. 
set your dark theme time 3 minutes ahead of your **current time**

I meant that if it if your **current** time is 10pm, you should set the sunset time to 10:03pm, such that you only have to wait 1 minute, not until sunset.

**Comment by RayDe97** [comment]
Sorry I was out anyways. It worked, when I got back, I saw it had changed. I tested again with specific time and it worked too.

**Comment by Spiritreader** [comment]
Could you still attach the log?
I assume something is going on with the gpu module and I would like to investigate if it is a bug with auto dark Mode or your operating system not giving adm permission to read gpu usage.

**Comment by Spiritreader** [comment]
Thank you!
It looks like there is indeed an issue with the GPU module we can fix.
I will take a look and open a new issue if the solution proves to be more difficult.

**Comment by RayDe97** [comment]
Awesome happy to help

**Comment by Spiritreader** [comment]
Hmm actually, I am sorry, I may have blamed the module too quickly. It looks looks like your windows performance counter store got corrupted.

If you want to restore the functionality of the gpu module, you can attempt rebuilding your performance counters.

Tobii (yes the ones who make eye trackers, have a great resource on how to do this) 
https://connect.tobiipro.com/s/article/What-should-I-do-is-I-get-Cannot-load-Counter-Name-data-because-an-invalid-index-error?language=en_US
